profiler/data_collector.rst

```diff
@@ -73,6 +78,11 @@ The getters are added to give the template access to the collected information.
 
 .. caution::
 
+    If the data that is not directly related to the request or response, you need to make the data accessible to your DataCollector.
+    This can be achieved by injecting the service that holds the information you intend to profile into your DataCollector.
```

Comment:
I don't think this is needed, this note sounds clear enough, also a constructor is not this only way to inject data.
Note that the collector itself can be used as listener in order to grab anything, and then collect data using it, both injection and observation are used in the core btw.